### PR TITLE
Hw1task2

### PR DESCRIPTION
**Pull request by mrsebo67**

### CODE REVIEW

File: homework_1/task_2/main.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	amountToTax := 7000.0
+
+	tax, err := taxAPI.GetTax(taxBrackets, amountToTax, lastTaxRate)
+
+	if err != nil {
+		log.Fatal(
+			errors.WithMessage(err, "error in tax calculation"),
+		)
+	}
+	log.Println(tax)
```
Comment (oreskdaxilis) [inline homework_1/task_2/main.go:22]: Same comment regarding logging of the result as in task1.

File: homework_1/task_2/taxAPI/taxAPI.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+func GetTax(taxBrackets []TaxBracket, amountToTax float64, lastRate float64) (float64, error) {
+
+	err := CheckTaxBrackets(taxBrackets)
+	if err != nil {
+		return 0.0, err
+	}
+
+	var tax float64
+	var oldAmount float64 = 0
```
Comment (oreskdaxilis) [inline homework_1/task_2/taxAPI/taxAPI.go:42]: This is not needed; `0` is the default value.

File: homework_1/task_2/taxAPI/taxAPI.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+	return nil
+
+}
+
+func GetTax(taxBrackets []TaxBracket, amountToTax float64, lastRate float64) (float64, error) {
+
+	err := CheckTaxBrackets(taxBrackets)
+	if err != nil {
+		return 0.0, err
```
Comment (oreskdaxilis) [inline homework_1/task_2/taxAPI/taxAPI.go:38]: You can just return `0` instead of `0.0` - the function is returning `float64` so there is no risk of accidentally casting the value to `int` (your approach is not wrong, it's just not usual).

File: homework_1/task_2/taxAPI/taxAPI.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
+package taxAPI
```
Comment (oreskdaxilis) [inline homework_1/task_2/taxAPI/taxAPI.go:1]: Package naming should be lowercase. More info here https://blog.golang.org/package-names

File: homework_1/task_2/main.go [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+	"log"
+	"task_2/taxAPI"
+
+	"github.com/pkg/errors"
+)
+
+func main() {
+	taxBrackets := []taxAPI.TaxBracket{{0, 2000}, {0.1, 7000}, {0.2, 10000}}
+	lastTaxRate := 0.3
+	amountToTax := 7000.0
```
Comment (oreskdaxilis) [inline homework_1/task_2/main.go:13]: It would be nice to use `amountToTax` as an argument to the program (it would be even better to have tax brackets configurable as well, but providing amount to the program should be fairly simple).

File: homework_1/task_2/taxAPI/testCases_test.go [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+type testCase struct {
+	taxBrackets []taxAPI.TaxBracket
+	lastRate    float64
+	amountToTax float64
+
+	expectedOutput float64
+	expectingError bool
+}
+
+func getTestCases() []testCase {
```
Comment (oreskdaxilis) [inline homework_1/task_2/taxAPI/testCases_test.go:14]: Consider adding few more test cases - there are many different edge cases here (negative/zero amountToTax, negative/zero lastTaxRate, having empty tax brackets, having tax brackets where values are not correctly sorted etc); no need to cover absolutely everything, but few more test cases would be handy.
